test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -509,11 +571,33 @@ def skip_pt20_cuda121_tests(
 
 
 @pytest.fixture(autouse=True)
-def skip_p5_tests(instance_type, processor, ecr_image):
-    if "p5." in instance_type:
-        image_cuda_version = get_cuda_version_from_tag(ecr_image)
-        if processor != "gpu" or Version(image_cuda_version.strip("cu")) < Version("120"):
-            pytest.skip("Images using less than CUDA 12.0 doesn't support P5 EC2 instance.")
+def skip_pt21_test(request):
+    if "framework_version" in request.fixturenames:
+        fw_ver = request.getfixturevalue("framework_version")
+    elif "ecr_image" in request.fixturenames:
+        fw_ver = request.getfixturevalue("ecr_image")
+    else:
+        return
+    if request.node.get_closest_marker("skip_pt21_test"):
+        if Version(fw_ver) in SpecifierSet("==2.1"):
+            pytest.skip(
+                f"PT2.1 SM DLC doesn't support SMDMP and SMDDP for now, so skipping this container with tag {fw_ver}"
```

Comment:
nit: This is an unclear skip message based on the fixture name, as the fixture implies that it can be used for any pt21 skip condition, but the message is coupled to SMDDP and SMMP.